### ISSUE TITLE
ESP32 samples/net/wifi DHCP fails

### ISSUE THREAD

**Issue by svenschwermer** [zephyrproject-rtos/zephyr#55834]
I have built samples/net/wifi for esp32 with the following extra config:
```
CONFIG_ESP32_WIFI_STA_AUTO_DHCPV4=y
CONFIG_WIFI_LOG_LEVEL_DBG=y
CONFIG_NET_DHCPV4_LOG_LEVEL_DBG=y
```
Flashing this yields:
```
% west flash && west espressif monitor
-- west flash: rebuilding
ninja: no work to do.
-- west flash: using runner esp32
-- runners.esp32: Flashing esp32 chip on None (921600bps)
esptool.py v3.3
Found 1 serial ports
Serial port /dev/ttyUSB0
Connecting....
Detecting chip type... Unsupported detection protocol, switching and trying again...
Connecting.....
Detecting chip type... ESP32
Chip is ESP32-D0WDQ6 (revision 1)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: 24:0a:c4:2f:9e:e0
Uploading stub...
Running stub...
Stub running...
Changing baud rate to 921600
Changed.
Configuring flash size...
Auto-detected Flash size: 4MB
Flash will be erased from 0x00001000 to 0x00007fff...
Flash will be erased from 0x00008000 to 0x00008fff...
Flash will be erased from 0x00010000 to 0x0007efff...
Flash params set to 0x0220
Wrote 32768 bytes at 0x00001000 in 0.6 seconds (440.0 kbit/s)...
Hash of data verified.
Wrote 16384 bytes at 0x00008000 in 0.3 seconds (493.3 kbit/s)...
Hash of data verified.
Wrote 458752 bytes at 0x00010000 in 6.5 seconds (563.5 kbit/s)...
Hash of data verified.

Leaving...
Hard resetting via RTS pin...
Serial port /dev/ttyUSB0
Connecting....
Detecting chip type... Unsupported detection protocol, switching and trying again...
Connecting......
Detecting chip type... ESP32
--- idf_monitor on /dev/ttyUSB0 115200 ---
--- Quit: Ctrl+] | Menu: Ctrl+T | Help: Ctrl+T followed by Ctrl+H ---
ets Jun  8 2016 00:22:57
rst:0x1 (POWERON_RESET),boot:0x17 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:6596
load:0x40078000,len:15216
ho 0 tail 12 room 4
load:0x40080400,len:3992
entry 0x40080634
I (29) boot: ESP-IDF d2564ca5c 2nd stage bootloader
I (29) boot: compile time 21:56:27
I (29) boot: chip revision: 1
I (32) boot_comm: chip revision: 1, min. bootloader chip revision: 0
I (39) boot.esp32: SPI Speed      : 40MHz
I (44) boot.esp32: SPI Mode       : DIO
I (48) boot.esp32: SPI Flash Size : 4MB
I (53) boot: Enabling RNG early entropy source...
I (58) boot: Partition Table:
I (62) boot: ## Label            Usage          Type ST Offset   Length
I (69) boot:  0 nvs              WiFi data        01 02 00009000 00006000
I (76) boot:  1 phy_init         RF data          01 01 0000f000 00001000
I (84) boot:  2 factory          factory app      00 00 00010000 00100000
I (91) boot: End of partition table
I (95) boot_comm: chip revision: 1, min. application chip revision: 0
I (103) esp_image: segment 0: paddr=00010020 vaddr=00000020 size=0001ch (    28)
I (111) esp_image: segment 1: paddr=00010044 vaddr=3ffb0000 size=0175ch (  5980) load
I (122) esp_image: segment 2: paddr=000117a8 vaddr=3ffb175c size=003c8h (   968) load
I (128) esp_image: segment 3: paddr=00011b78 vaddr=3ffb1b28 size=00224h (   548) load
I (137) esp_image: segment 4: paddr=00011da4 vaddr=40080000 size=09518h ( 38168) load
I (161) esp_image: segment 5: paddr=0001b2c4 vaddr=00000000 size=04d74h ( 19828)
I (168) esp_image: segment 6: paddr=00020040 vaddr=3f400040 size=0ece4h ( 60644) map
I (191) esp_image: segment 7: paddr=0002ed2c vaddr=00000000 size=012ech (  4844)
I (193) esp_image: segment 8: paddr=00030020 vaddr=400d0020 size=4e5bch (320956) map
I (318) boot: Loaded app from partition at offset 0x10000
I (318) boot: Disabling RNG early entropy source...

[00:00:02.270,000] <dbg> net_dhcpv4: net_dhcpv4_init: (main):
*** Booting Zephyr OS build zephyr-v3.3.0-1032-g332f326ab18b ***
uart:~$ wifi connect ***REDACTED*** 0 ***REDACTED***
Connection requested
[00:00:17.546,000] <dbg> net_dhcpv4: dhcpv4_start_internal: (esp_event): iface 0x3ffb1b28 state=init
[00:00:17.546,000] <dbg> net_dhcpv4: dhcpv4_start_internal: (esp_event): wait timeout=7s
[00:00:17.546,000] <dbg> net_dhcpv4: dhcpv4_set_timeout: (esp_event): sched timeout dhcvp4=0x3ffb1bc8 timeout=7s
[00:00:17.546,000] <dbg> net_dhcpv4: dhcpv4_manage_timers: (sysworkq): iface 0x3ffb1b28 dhcpv4=0x3ffb1bc8 state=init timeleft=7
[00:00:17.546,000] <dbg> net_dhcpv4: dhcpv4_timeout: (sysworkq): Waiting for 7s
[00:00:24.546,000] <dbg> net_dhcpv4: dhcpv4_manage_timers: (sysworkq): iface 0x3ffb1b28 dhcpv4=0x3ffb1bc8 state=init timeleft=0
[00:00:24.546,000] <dbg> net_dhcpv4: dhcpv4_enter_selecting: (sysworkq): enter state=selecting
[00:00:24.548,000] <dbg> esp32_wifi: esp32_wifi_send: pkt sent 0x3fff8324 len 291
[00:00:24.548,000] <dbg> net_dhcpv4: dhcpv4_set_timeout: (sysworkq): sched timeout dhcvp4=0x3ffb1bc8 timeout=3s
[00:00:24.548,000] <dbg> net_dhcpv4: dhcpv4_send_discover: (sysworkq): send discover xid=0x1ba49110 timeout=3s
[00:00:24.548,000] <dbg> net_dhcpv4: dhcpv4_timeout: (sysworkq): Waiting for 3s
[00:00:24.548,000] <dbg> net_dhcpv4: dhcpv4_manage_timers: (sysworkq): iface 0x3ffb1b28 dhcpv4=0x3ffb1bc8 state=selecting timeleft=3
[00:00:24.548,000] <dbg> net_dhcpv4: dhcpv4_timeout: (sysworkq): Waiting for 3s
[00:00:27.548,000] <dbg> net_dhcpv4: dhcpv4_manage_timers: (sysworkq): iface 0x3ffb1b28 dhcpv4=0x3ffb1bc8 state=selecting timeleft=0
[00:00:27.549,000] <dbg> esp32_wifi: esp32_wifi_send: pkt sent 0x3fff8324 len 291
[00:00:27.549,000] <dbg> net_dhcpv4: dhcpv4_set_timeout: (sysworkq): sched timeout dhcvp4=0x3ffb1bc8 timeout=7s
[00:00:27.549,000] <dbg> net_dhcpv4: dhcpv4_send_discover: (sysworkq): send discover xid=0x1ba49111 timeout=7s
[00:00:27.549,000] <dbg> net_dhcpv4: dhcpv4_timeout: (sysworkq): Waiting for 7s
[00:00:27.549,000] <dbg> net_dhcpv4: dhcpv4_manage_timers: (sysworkq): iface 0x3ffb1b28 dhcpv4=0x3ffb1bc8 state=selecting timeleft=7
[00:00:27.549,000] <dbg> net_dhcpv4: dhcpv4_timeout: (sysworkq): Waiting for 7s
[00:00:34.550,000] <dbg> net_dhcpv4: dhcpv4_manage_timers: (sysworkq): iface 0x3ffb1b28 dhcpv4=0x3ffb1bc8 state=selecting timeleft=0
[00:00:34.551,000] <dbg> esp32_wifi: esp32_wifi_send: pkt sent 0x3fff8324 len 291
[00:00:34.551,000] <dbg> net_dhcpv4: dhcpv4_set_timeout: (sysworkq): sched timeout dhcvp4=0x3ffb1bc8 timeout=17s
[00:00:34.551,000] <dbg> net_dhcpv4: dhcpv4_send_discover: (sysworkq): send discover xid=0x1ba49112 timeout=17s
[00:00:34.551,000] <dbg> net_dhcpv4: dhcpv4_timeout: (sysworkq): Waiting for 17s
[00:00:34.551,000] <dbg> net_dhcpv4: dhcpv4_manage_timers: (sysworkq): iface 0x3ffb1b28 dhcpv4=0x3ffb1bc8 state=selecting timeleft=17
[00:00:34.551,000] <dbg> net_dhcpv4: dhcpv4_timeout: (sysworkq): Waiting for 17s
[00:00:51.551,000] <dbg> net_dhcpv4: dhcpv4_manage_timers: (sysworkq): iface 0x3ffb1b28 dhcpv4=0x3ffb1bc8 state=selecting timeleft=0
[00:00:51.553,000] <dbg> esp32_wifi: esp32_wifi_send: pkt sent 0x3fff8324 len 291
[00:00:51.553,000] <dbg> net_dhcpv4: dhcpv4_set_timeout: (sysworkq): sched timeout dhcvp4=0x3ffb1bc8 timeout=32s
[00:00:51.553,000] <dbg> net_dhcpv4: dhcpv4_send_discover: (sysworkq): send discover xid=0x1ba49113 timeout=32s
[00:00:51.553,000] <dbg> net_dhcpv4: dhcpv4_timeout: (sysworkq): Waiting for 32s
[00:00:51.553,000] <dbg> net_dhcpv4: dhcpv4_manage_timers: (sysworkq): iface 0x3ffb1b28 dhcpv4=0x3ffb1bc8 state=selecting timeleft=32
[00:00:51.553,000] <dbg> net_dhcpv4: dhcpv4_timeout: (sysworkq): Waiting for 32s
[ ... in continues like this ... ]
```
I have found the following possibly related issue: https://github.com/zephyrproject-rtos/zephyr/issues/36971

I'm on zephyr-v3.3.0-1032-g332f326ab1 with hal-espressif@d2564ca5c14439888f7c4351884a2cf4db517a33

**Comment by svenschwermer** [comment]
> @svenschwermer Could you give #55854 a try?

That seems to fix the issue, that's for the very swift fix!

**Comment by mniestroj** [comment]
I'll reopen it for now. It will be closed automatically once #55854 gets merged.

Thanks @svenschwermer for testing!
